File: KnowledgeWithoutBorders/System_KWB/frmSplash.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace System_KWB
{
    public partial class frmSplash : Form
    {
        public frmSplash()
        {
            InitializeComponent();
        }

        private void Temporizador_Tick(object sender, EventArgs e)
        {
            if (Progresso.Value == 25)
            {
                Descrição.Text = "Configurando os Formulários...";
            }

            if (Progresso.Value == 50)
            {
                Descrição.Text = "Atualizando o Banco de Dados...";
            }

            if (Progresso.Value == 75)
            {
                Descrição.Text = "Carregando Ferramentas...";
            }

            if (Progresso.Value < Progresso.Maximum)
            {
                Progresso.Increment(5);
            }
            else
            {
                Temporizador.Stop();
                this.Hide();
                frmLogin Login = new frmLogin();
                Login.Show();

            //  frmPrincipal Principal = new frmPrincipal();
            //  Principal.Show();
            }
        }

        private void pictureCancelar_Click(object sender, EventArgs e)
        {
            Temporizador.Stop();

            DialogResult opcaoSelecionada;
            opcaoSelecionada = MessageBox.Show("Deseja realmente sair do sistema?", "Mensagem do Sistema", MessageBoxButtons.YesNo, MessageBoxIcon.Question);
            if (opcaoSelecionada.Equals(DialogResult.No))
            {
                Temporizador.Start();
            }
            else
            {
                Application.Exit();
            }
        }
    }
}

File: OtakuFest/OtakuFest/frmSplash.vb
Public Class frmSplash

    Private Sub Temporizador_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Temporizador.Tick
        If Sub_Progresso.Value < Sub_Progresso.Maximum Then
            Sub_Progresso.Increment(5)
        ElseIf (Progresso.Value < Progresso.Maximum) Then
            Sub_Progresso.Value = 0
            Progresso.Increment(25)
        End If

        Select Case (Progresso.Value)
            Case 25
                Mensagem.Text = "Carregando o banco de dados..."
            Case 50
                Mensagem.Text = "Organizando ferramentas..."
            Case 75
                Mensagem.Text = "Inicializando o sistema..."
        End Select

        If (Progresso.Value = Progresso.Maximum) Then
            Temporizador.Stop()
            Me.Hide()
            frmLogin.ShowDialog()
        End If
    End Sub
End Class
